File: leetcode/python/2486_append_characters_to_string_to_make_subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        n = self.lcs(s, t)
        return len(t) - n

    def lcs(self, s: str, t: str) -> int:
        i = 0
        j = 0
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                j += 1
            i += 1
        return j

File: leetcode/rust/src/2486_append_characters_to_string_to_make_subsequence.rs
impl Solution {
    pub fn append_characters(s: String, t: String) -> i32 {
        let s_chars: Vec<char> = s.chars().collect();
        let t_chars: Vec<char> = t.chars().collect();
        let mut i = 0;
        let mut j = 0;

        while i < s_chars.len() && j < t_chars.len() {
            if s_chars[i] == t_chars[j] {
                j += 1;
            }
            i += 1;
        }
        return t.len() as i32 - j as i32;
    }
}
